src/process/volcanic/CCPr_Volcanic_Mod.F90

```diff
@@ -0,0 +1,381 @@
+         VolcanicState%nVolcanicSpecies = EmisState%Cats(VolcanicState%CatIndex)%nSpecies
+
+         !------------------------------------
+         ! Allocate emission species index
+         ALLOCATE( VolcanicState%VolcanicSpeciesIndex(VolcanicState%nVolcanicSpecies), STAT=RC )
+         CALL CC_CheckVar('VolcanicState%VolcanicSpeciesIndex', 0, RC)
+         IF (RC /= CC_SUCCESS) RETURN
+         VolcanicState%VolcanicSpeciesIndex = -1
+
+         ! Allocate emission speceis names
+         ALLOCATE( VolcanicState%VolcanicSpeciesName(VolcanicState%nVolcanicSpecies), STAT=RC )
+         CALL CC_CheckVar('VolcanicState%VolcanicSpeciesName', 0, RC)
+         IF (RC /= CC_SUCCESS) RETURN
+         VolcanicState%VolcanicSpeciesName = ''
+
+         ! Allocate CatChem species index
+         ALLOCATE( VolcanicState%SpcIDs(VolcanicState%nVolcanicSpecies), STAT=RC )
+         CALL CC_CheckVar('VolcanicState%SpcIDs', 0, RC)
+         IF (RC /= CC_SUCCESS) RETURN
+         VolcanicState%SpcIDs = -1
+
+         ! Allocate emission flux
+         ALLOCATE( VolcanicState%EmissionPerSpecies(VolcanicState%nVolcanicSpecies, &
+            SIZE(EmisState%Cats(VolcanicState%CatIndex)%Species(1)%Flux)), STAT=RC )
+         CALL CC_CheckVar('VolcanicState%EmissionPerSpecies', 0, RC)
+         IF (RC /= CC_SUCCESS) RETURN
+         VolcanicState%EmissionPerSpecies = ZERO
+
+         ! Allocate total emissions
+         ALLOCATE( VolcanicState%TotalEmission(SIZE(EmisState%Cats(VolcanicState%CatIndex)%Species(1)%Flux)), STAT=RC )
+         CALL CC_CheckVar('VolcanicState%TotalEmission', 0, RC)
+         IF (RC /= CC_SUCCESS) RETURN
+         VolcanicState%TotalEmission = ZERO
+
+         ! Set the file directory
+         VolcanicState%FileDir = TRIM(Config%Volcanic_filedir)
+
+      else
+         VolcanicState%Activate = .false.
+      end if
+
+   end subroutine CCPR_Volcanic_Init
+
+   !>
+   !! \brief Run the VolcanicEmissions
+   !!
+   !! \param [IN] MetState - The MetState object
+   !! \param [INOUT] VolcanicState - The VolcanicState object
+   !! \param [INOUT] EmisState - The EmisState object
+   !! \param [INOUT] RC Return code
+   !!
+   !! \ingroup catchem_Volcanicemissions_process
+   !!!>
+   SUBROUTINE CCPr_Volcanic_Run( MetState, VolcanicState, EmisState, RC )
+
+      ! USE
+      USE constants, only : g0
+      use CCPr_Scheme_Volcanic_GOCART_Mod, only : CCPr_Scheme_Volcanic_GOCART, VolcanicEmisData, ReadASCIIPointEmis
+
+      IMPLICIT NONE
+      ! INPUT PARAMETERS
+      TYPE(MetStateType),  INTENT(IN) :: MetState       !< MetState Instance
+
+      ! INPUT/OUTPUT PARAMETERS
+      !TYPE(DiagStateType), INTENT(INOUT)      :: DiagState       !< DiagState Instance
+      TYPE(VolcanicStateType), INTENT(INOUT)  :: VolcanicState  !< VolcanicState Instance
+      !TYPE(ChemStateType), INTENT(INOUT)     :: ChemState       !< ChemState Instance
+      TYPE(EmisStateType), INTENT(INOUT)     :: EmisState       !< ChemState Instance
+
```

Comment:
Would we ever need the DiagState or ChemState here? If not, we should delete for clarity.